auth-web/src/services/business-lookup.services.ts

```diff
@@ -0,0 +1,62 @@
+import { BusinessLookupResultIF } from '@/models'
+import ConfigHelper from '@/util/config-helper'
+import { SessionStorageKeys } from '@/util/constants'
+import { axios } from '@/util/http-util'
+
+/**
+ * Class that provides integration with the BusinessLookup API.
+ * This class is substantially similar to the same class in Create and Edit UIs.
+ */
+export default class BusinessLookupServices {
+  /** The full Registries Search API URL. */
+  static get registriesSearchApiUrl (): string {
+    const url = `${process.env.VUE_APP_REGISTRIES_SEARCH_API_URL}`
+    const version = `${process.env.VUE_APP_REGISTRIES_SEARCH_API_VERSION}`
+    return (url + version + '/')
+  }
+
+  /** The Registries Search API Key. */
+  static get registriesSearchApiKey (): string {
+    return `${process.env.VUE_APP_REGISTRIES_SEARCH_API_KEY}`
+  }
+
+  /** The Account ID. */
+  static get accountId (): string {
+    const account = JSON.parse(
+      ConfigHelper.getFromSession(SessionStorageKeys.CurrentAccount || '{}')
+    )
+    return account?.id
+  }
+
+  /**
+   * Searches for business by code or words.
+   * @param query code or words to search
+   * @param status status to match (ACTIVE or HISTORICAL or empty to match all statuses)
+   * @returns a promise to return the search results
+   */
+  static async search (query: string, status = ''): Promise<BusinessLookupResultIF[]> {
+    const legalType = 'BC,A,ULC,C,S,XP,GP,LP,CUL,XS,LLC,LL,BEN,CP,CC,XL,FI,XCP,PA'
+
+    let url = this.registriesSearchApiUrl + 'businesses/search/facets?start=0&rows=20'
+    url += `&categories=legalType:${legalType}${status ? '::status:' + status : ''}`
+    url += `&query=value:${encodeURIComponent(query)}`
+
+    return axios.get(url, {
```

Comment:
@seeker25 @kialj876 @argush3 @vysakh-menon-aot 

I'm wondering about the near future, where we need to implement a NR Search service and API ([ticket 16714](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/16714)), which will likely be implemented in Namex API...

And I'm wondering about the need to have config keys for the URL and API key...

Do we want some consistency/symmetry between the Business Search calls and the NR Search calls? Would it make sense to proxy one or both via Auth API?

Reply:
I don't think it makes sense to do add an endpoint in auth that just relays data everyone can already access directly. Both calls are fully public so there's no reason to verify the token either (if we needed that then we would have to do it in the search-api/namex-api as well anyways).

Reply:
OK, thanks.

Do you think we'll need to verify the token (or key) for a NR search via SOLR?

Reply:
I think this conversation is complete (but feel free to add anything pertinent).

A follow-up conversation is continuing in the [NR search API ticket](https://github.com/bcgov/entity/issues/16714).